rmw_zenoh_cpp/src/detail/rmw_publisher_data.cpp

```diff
@@ -355,19 +319,18 @@ rmw_ret_t PublisherData::publish_serialized_message(
   // The encoding is simply forwarded and is useful when key expressions in the
   // session use different encoding formats. In our case, all key expressions
   // will be encoded with CDR so it does not really matter.
-  z_publisher_put_options_t options;
-  z_publisher_put_options_default(&options);
-  uint8_t local_gid[RMW_GID_STORAGE_SIZE];
-  entity_->copy_gid(local_gid);
-  z_owned_bytes_t attachment;
-  create_map_and_set_sequence_num(&attachment, sequence_number_++, local_gid);
-  options.attachment = z_move(attachment);
-  z_owned_bytes_t payload;
-  z_bytes_copy_from_buf(&payload, serialized_message->buffer, data_length);
-
-  z_result_t res = z_publisher_put(z_loan(pub_), z_move(payload), &options);
-  if (res != Z_OK) {
-    if (res == Z_ESESSION_CLOSED) {
+  zenoh::ZResult result;
+  auto options = zenoh::Publisher::PutOptions::create_default();
+  options.attachment = create_map_and_set_sequence_num(sequence_number_++, entity_->copy_gid());
+
+  std::vector<uint8_t> raw_image(
+    serialized_message->buffer,
+    serialized_message->buffer + data_length);
+  zenoh::Bytes payload(raw_image);
```

Comment:
I have a minor and a major question to ask here.

The minor thing is that I don't love the term "raw_image" here, as this is any data, not just images.  Maybe "raw_data" instead?

The more major thing I want to ask about this is how many copies we are doing with this sequence.  Does constructing a `std::vector` with a pointer and length make a copy, or just refer to the pointer?  And then does passing a `std::vector` into the `zenoh::Bytes` constructor take a reference, or make a copy?